### ISSUE TITLE
Unable to set a newly created flow in First Login Flow override for a SAML identity provider

### ISSUE THREAD

**Issue by mitrx** [keycloak/keycloak#27691]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

I am trying to create a flow that will be used as First Login Flow for certain identity providers, but if I create a flow it doesn't show up in the selector of flows in identity provider configuration view.

### Version

24.0.1

### Regression

- [X] The issue is a regression

### Expected behavior

Any flow present in "Authentication" page can be selected as First Login Flow override in identity provider configuration view.

### Actual behavior

The flows that existed before (in my case 23.0.7 version) are present in First Login Flow selector list, but newly created flows do not appear in the selector list.

The help tooltip next to the selector is also not showing the text (compared to 23.0.7 version).

### How to Reproduce?

1. Create a flow in "Authentication" page.
2. Go to "Identity providers" -> \<some SAML identity provider\> -> open selector next to "First login flow override" label.
3. The flow created at step 1 doesn't show up in the list.

**Comment by mitrx** [comment]
My guess is that the problem is in the creation of a flow, because rolling back to 23.0.7 does not make a flow appear in the list.

**Comment by hmlnarik** [comment]
@mitrx Thank you for reporting the issue. I was unable to reproduce the issue, after creating a basic flow, the new flow was present in the _First login flow override_ field. I had to scroll down the list though using cursor keys or a mouse wheel. Could this be the cause?

Right after opening the _First login flow override_ combo box:
![image](https://github.com/keycloak/keycloak/assets/5391360/9cfde07c-033a-41dc-a5fa-deef29dd369c)
After scrolling down (note the _my flow_ fresh flow):
![image](https://github.com/keycloak/keycloak/assets/5391360/1c9e5d7d-73fa-4f26-9391-95ccc1f1c650)

**Comment by mitrx** [comment]
@hmlnarik I checked again and what I am seeing is that the flows are being fetched in the Network tab of the browser and my custom flow is present in the response. However, I checked from multiple browsers and then in the DOM and I couldn't find my custom flow in the list.
I tested on a Keycloak instance deployed via docker as suggested in the [getting started page](https://www.keycloak.org/getting-started/getting-started-docker).
My browser is `Mozilla Firefox 123.0.1`
Attaching the screenshots of how it looks for me:
![Screenshot 2024-03-11 at 14-16-39 Keycloak Administration UI](https://github.com/keycloak/keycloak/assets/8896194/3da4b4fb-3539-4f61-ae0e-155472e2870d)
![Screenshot 2024-03-11 at 14-17-08 Keycloak Administration UI](https://github.com/keycloak/keycloak/assets/8896194/28dcf3da-250a-4d1a-b1bc-cdd237387529)
The list in my DOM:
```html
<ul class="pf-c-select__menu" role="listbox" aria-label="First login flow override" required="">
  <li id="select-option-1710163266366a06d47mcxge" role="presentation" class="pf-c-select__menu-wrapper">
    <button class="pf-c-select__menu-item" role="option" type="button">
    </button>
  </li>
  <li id="select-option-1710163266366u2ilnj52qm" role="presentation" class="pf-c-select__menu-wrapper">
    <button class="pf-c-select__menu-item" role="option" type="button">browser
    </button>
  </li>
  <li id="select-option-1710163266366yulpmw0nxkd" role="presentation" class="pf-c-select__menu-wrapper">
    <button class="pf-c-select__menu-item" role="option" type="button">direct grant
    </button>
  </li>
  <li id="select-option-1710163266366npyfrszu0t" role="presentation" class="pf-c-select__menu-wrapper">
    <button class="pf-c-select__menu-item" role="option" type="button">registration
    </button>
  </li>
  <li id="select-option-1710163266366zp5p3tz01j" role="presentation" class="pf-c-select__menu-wrapper">
    <button class="pf-c-select__menu-item" role="option" type="button">reset credentials
    </button>
  </li>
  <li id="select-option-1710163266366puchnf7ul8" role="presentation" class="pf-c-select__menu-wrapper">
    <button class="pf-c-select__menu-item" role="option" type="button">first broker login
    </button>
  </li>
  <li id="select-option-1710163266366i1odim5htj" role="presentation" class="pf-c-select__menu-wrapper">
    <button class="pf-c-select__menu-item" role="option" type="button">docker auth
    </button>
  </li>
</ul>
```
The JSON returned by `GET /admin/realms/master/authentication/flows`:
[flows.json](https://github.com/keycloak/keycloak/files/14559267/flows.json)

**Comment by hmlnarik** [comment]
Thank you. This is not a UI issue, since the flows seem to be created without correct provider (`"providerId": ""`) That is also the reason you do not see the flows in the UI. How have you created the flows? Did you use UI or admin CLI? Can you recreate the flows via the UI?

**Comment by mitrx** [comment]
Yes, I think I suggested this is the issue in one of my previous comments here https://github.com/keycloak/keycloak/issues/27691#issuecomment-1985031695.
I created the flows via UI and I did everything from scratch before writing my previous comment. I think it happens every time.
Also, note the tooltip issue. Maybe it is somehow related.

**Comment by ssilvert** [comment]
I tried it out and it is working for me as well.

**Comment by ssilvert** [comment]
@mitrx There isn't much we can do if we can't reproduce the problem on our end.  I'm hoping you can help us figure out what is different in your environment.

**Comment by anna-slapik** [comment]
I have the same issue: created simple login flow with single step as described here: https://ultimatesecurity.pro/post/okta-oidc/. Don't see the flow in the drop down. Run custom keycloak image with docker locally. Needed custom image to create custom mapper protocol as spi

**Comment by mitrx** [comment]
@ssilvert are you sure you used exactly the tag I reported and not on master or some other branch? It shouldn't be a browser specific issue and I cannot help much but saying I followed exactly the steps described in the getting started guide for docker and used the following command to run a new keycloak environment:
```
docker run -p 8080:8080 -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin quay.io/keycloak/keycloak:24.0.1 start-dev
```

I saw there was 24.0.2 released a few days ago. I made the same test on it and it is back working if you run a new container with keycloak 24.0.2. Then I tested again from scratch on 24.0.1 and it is not working on that tag. However, the tooltip is still broken from 23.0.7.

To sum up:
23.0.7: working list, working tooltip
24.0.1: broken list, broken tooltip
24.0.2: working list, broken tooltip

@anna-slapik I believe your problem should be resolved if you make sure you use 24.0.2 and then delete your custom flow and re-create it. After that I think it should appear in the list.

**Comment by anna-slapik** [comment]
@mitrx , update to 24.0.2 worked for me, thanks!

**Comment by ssilvert** [comment]
@mitrx I tried again too on a recent build and still working. 

I do see the tooltip problem.  I'll take this issue out of triage to address that issue.

**Comment by jonkoops** [comment]
@ssilvert is this something that needs to be backported?

**Comment by ssilvert** [comment]
> @ssilvert is this something that needs to be backported?

Not worth the effort.
